src/HttpMock/EndpointMatchingRule.cs

```diff
@@ -28,7 +29,14 @@ public class EndpointMatchingRule : IMatchingRule
 				queryParamMatch = new QueryParamMatch().MatchQueryParams(requestHandler, requestQueryParams);
 			}
 
-			return uriStartsWith && httpMethodsMatch && queryParamMatch;
+			bool headerMatch = true;
+			bool shouldMatchHeaders = (requestHandler.RequestHeaders.Count > 0);

```

Comment:
Add a null check here. This will avoid having to set the header dictionary in the unit tests.